onnx/shape_inference/implementation.cc

```diff
@@ -115,6 +115,42 @@ void checkShapesAndTypes(const TypeProto& inferredType, const TypeProto& existin
     fail_type_inference("type case unsupported. existing=", existingTypeCase, " inferred=", inferredTypeCase);
   }
 }
+// after symbolic shape_inference from inferredType, assign it to existingTensorType
+// TypeProto_Tensor or TypeProto_SparseTensor
+template <typename T>
+void generateSymbolicShape(T inferredType, T* existingTensorType, SymbolTable symbolTable) {
```

Comment:
this is very confusing... I don't understand what are you trying to do here...

All you need to do is check whether inferredType consists an unknown symbol, if so then generate a new symbol and set dim param... why do you need inferredType and existingTensorType.... + these names are misleading because both are same based on line 142

Reply:
SymbolTable& symbolTable

Reply:
this method should be simplified

Reply:
Thanks for catching this @askhade and @gramalingam ! I have simplified the arguments for `generateSymbolicShape` and `materializeSymbolicShape` by only using mutable_XXX_type()